services/payments/datastore.go

```diff
@@ -156,11 +156,24 @@ func (q *QLDBDatastore) UpdatePaymentState(ctx context.Context, documentID strin
 		ctx,
 		func(txn qldbdriver.Transaction) (interface{}, error) {
 			_, err := txn.Execute(
-				"UPDATE transactions BY d_id SET data = ?, signature = ?, publicKey = ? WHERE d_id = ?",
+				`UPDATE transactions BY d_id
+						SET
+							data = ?,
+							signature = ?,
+							publicKey = ?
+						WHERE
+							d_id = ?
+							AND data != ?,
+							AND signature != ?,
```

Comment:
I'm worried that this won't work in practice - are the nitro attestations fully deterministic?

Reply:
Hm. Not sure, honestly. What do you think of just doing this comparison against `data`?